editoast/src/views/path/path_item_cache.rs

```diff
@@ -173,6 +205,34 @@ impl PathItemCache {
 
         Ok(result)
     }
+
+    /// Filter operational points parts by a track label or a track id
+    /// If neither a track label or id is provided, the original list is returned
+    fn track_reference_filter(
+        &self,
+        track_offsets: Vec<TrackOffset>,
+        track_reference: &Option<TrackReference>,
+    ) -> Vec<TrackOffset> {
+        if let Some(track_reference) = track_reference {
```

Comment:
nit: you could do a `match` for the original `track_reference`, it would make clear when reading the function that there is 3 cases to handle (instead of 1 case, then another nested 1 case).

```rust
match track_reference {
  Some(TrackReference::Id { track_id }) => /* ... */,
  Some(TrackReference::Label { track_label }) => /* ... */,
  None => /* ... */,
```